pkg/cmd/server/apis/config/types.go

```diff
@@ -1757,3 +1757,84 @@ type ServerProjectConfig struct {
 	// If it is not specified, a default template is used.
 	ProjectRequestTemplate string
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type KubeAPIServerConfig struct {
+	metav1.TypeMeta
+
+	// ServingInfo describes how to start serving
+	ServingInfo HTTPServingInfo
+
+	// CORSAllowedOrigins
+	CORSAllowedOrigins []string
+
+	// OAuthConfig, if present start the /oauth endpoint in this process
+	OAuthConfig *OAuthConfig
+
+	// AuthConfig configures authentication options in addition to the standard
+	// oauth token and client certificate authenticators
+	AuthConfig MasterAuthConfig
+
+	// AggregatorConfig has options for configuring the aggregator component of the API server.
+	AggregatorConfig AggregatorConfig
+
+	AuditConfig AuditConfig
+
+	// StoragePrefix is the path within etcd that the OpenShift resources will
+	// be rooted under. This value, if changed, will mean existing objects in etcd will
+	// no longer be located.
+	StoragePrefix string
+
+	// EtcdClientInfo contains information about how to connect to etcd
+	EtcdClientInfo EtcdConnectionInfo
+
+	// KubeletClientInfo contains information about how to connect to kubelets
+	KubeletClientInfo KubeletConnectionInfo
+
+	AdmissionPluginConfig map[string]*AdmissionPluginConfig
+
+	// ServicesSubnet is the subnet to use for assigning service IPs
+	ServicesSubnet string
+	// ServicesNodePortRange is the range to use for assigning service public ports on a host.
+	ServicesNodePortRange string
+
+	// TODO this should be removable in a later release after we've completed migration
+	LegacyServiceServingCertSignerCABundle string
+
+	// UserAgentMatchingConfig controls how API calls from *voluntarily* identifying clients will be handled.  THIS DOES NOT DEFEND AGAINST MALICIOUS CLIENTS!
+	// TODO I think we should just drop this feature.
+	UserAgentMatchingConfig UserAgentMatchingConfig
+
+	ImagePolicyConfig KubeAPIServerImagePolicyConfig
```

Comment:
doesn't think belong to openshift api server?

Reply:
> doesn't think belong to openshift api server?

imagepolicyadmission uses it.